src/adapters/scene.js

```diff
@@ -11,21 +11,33 @@ import loadImage from '../libs/image_loader'
 import nconf from "../../local_modules/nconf_getter"
 import MapPoi from "./poi/map_poi";
 import HotLoadPoi from "./poi/hotload_poi";
+import Store from '../adapters/store'
 
 const baseUrl = nconf.get().system.baseUrl
+const store = new Store()
 
 function Scene() {
+  this.currentMarker = null
+  this.popup = new PoiPopup()
+}
+
+Scene.prototype.initScene = async function () { 
   UrlState.registerHash(this, 'map')
-  if(window.hotLoadPoi) {
+  if (window.hotLoadPoi) {
```

Comment:
you should regroup all init position logic and put it inside a method called setupMapPosition or something equivalent.